controllers/vspherevm_controller.go

```diff
@@ -279,6 +279,13 @@ func (r vmReconciler) Reconcile(ctx goctx.Context, req ctrl.Request) (_ ctrl.Res
 		}
 	}
 
+	// If the VSphereVM doesn't have our finalizer, add it.
+	// Requeue immediately to avoid the race condition between init and delete
+	if !ctrlutil.ContainsFinalizer(vsphereVM, infrav1.VMFinalizer) {
+		ctrlutil.AddFinalizer(vsphereVM, infrav1.VMFinalizer)
+		return reconcile.Result{}, nil
+	}
+
```

Comment:
```suggestion
	if vsphereVM.ObjectMeta.DeletionTimestamp.IsZero() {
		// If the VSphereVM doesn't have our finalizer, add it.
		// Requeue immediately to avoid the race condition between init and delete
		if !ctrlutil.ContainsFinalizer(vsphereVM, infrav1.VMFinalizer) {
			ctrlutil.AddFinalizer(vsphereVM, infrav1.VMFinalizer)
			return reconcile.Result{}, nil
		}
	}
```
In this case we have to add the if to ensure we only do this if the deletionTimestamp is not set, as the check for deletionTimestamp happens within reconcile

Reply:
Will fix it in next patch